### PR TITLE
Bug: is_identifier_string is broken for some characters under typical environments (e.g. "Terser on Node.js and run on browser")

### PR DESCRIPTION
**Pull request by naruaway**
**TL;DR; When running on Node.js (e.g. v18), `is_identifier_string` incorrectly decide that "`a・b` is a valid identifier" (the string includes "U+30FB KATAKANA MIDDLE DOT") and Terser applies optimization based on this and it can generate broken JavaScript for typical environments.**

For example, `console.log({"a・b":123})` will be `console.log({a・b:123})`, which will throw `SyntaxError: Invalid or unexpected token` when parsed on recent Google Chrome (tested v123). Note that we can check this "syntax error" also in [this TypeScript Playground](https://www.typescriptlang.org/play?ts=5.4.3#code/MYewdgziA2CmB00QHMAUBvAhob4YBGAuARgCYBmAXwEog).

## How does this happen?
[is_identifier_string](https://github.com/terser/terser/blob/1161ff1b6b254953bcb72d7dd6c661c26c6bdb6a/lib/parse.js#L351-L370) internally uses `ID_Continue` regex, which uses `\p{ID_Continue}` when available ([source](https://github.com/terser/terser/blob/1161ff1b6b254953bcb72d7dd6c661c26c6bdb6a/lib/parse.js#L265-L277)). This means that depending on the runtime environment (e.g. Node.js / Deno / Bun) and the version, the actual character set will be different since `ID_Continue` is being updated by Unicode spec.

[Unicode says](https://www.unicode.org/reports/tr31/#Backward_Compatibility):
> The Other_ID_Start and Other_ID_Continue properties are thus designed to ensure that the Unicode identifier specification is backward compatible. Any sequence of characters that qualified as an identifier in some version of Unicode will continue to qualify as an identifier in future versions

So if we use newer logic for deciding "is_identifier" for a minifier, it can generate JavaScript file which is not compatibile with older environment.
I think it's better to hard-code the whole logic of "is_identifier". Otherwise any runtime updates such as Node.js update can start generating yet another broken JavaScript from Terser.

## The solution in this PR

I just removed the variant to use unicode properties in regex. Notice that this bug is not reproducible when [running Terser in web browser](https://try.terser.org/) since `\p{ID_Continue}` is executed by browser, which has older set of characters than Node.js.
It would be nice if we can write test case to avoid reintroducing Unicode properties regex but so far I have no idea. I just added a test case for the currently known problematic case.

## Concerns for this PR

There might be performance degradation. But I think it's more important not to generate broken script for typical use cases.
Also it's better to make sure the behavior is consistent across environments for this kind of library IMHO

## Misc
`console.log(/\p{ID_Continue}/u.test('\u30FB'))` can be used for a quick test case, which can be `true` or `false` depending on supported unicode version

### CODE REVIEW

**Comment by fabiosantoscode** [comment]
Thanks for catching this!

**Comment by fabiosantoscode** [comment]
This looks good, thanks!

**Comment by naruaway** [comment]
Thank you so much for quick review & merge @fabiosantoscode!
I might write more details on this issue later on this MR or somewhere but we are mostly using Node.js v18 and v18.20.0 [got released recently](https://nodejs.org/en/blog/release/v18.20.0), which includes (for the official pre built binary):

> ICU updated to 74.2. Includes CLDR 44.1 and Unicode 15.1.

Here the ICU version update (74.2) was the trigger of the issue for "U+30FB KATAKANA MIDDLE DOT" character (other affected characters to be researched)

So after the Node.js update of v18.20.0, we observed some of the final JavaScript bundle from webpack got "broken" (I mean, it causes parse error even with recent Google Chrome)
Do you think you can publish the new version of `terser` soonish?

**Comment by fabiosantoscode** [comment]
As a mitigation for your issue, you can use the `ascii_only` format option (I think).

I release Terser on weekdays, around 1-3PM UTC, to avoid causing breakage at bad times.

**Comment by naruaway** [comment]
Ok, that's reasonable, thanks for clarification and your effort to maintain Terser 🙏

**Comment by naruaway** [comment]
Note: U+30FB and  U+0FF6 seem to have special historical context, which is specially treated and explained well in esbuild: https://github.com/evanw/esbuild/blob/44e746965d783646f97daf3d0617ff816727e7fb/scripts/gen-unicode-table.js#L51
